Add file function to async_impl::multipart

NaokiM03 opened the pull request:
Related to https://github.com/seanmonstar/reqwest/issues/646

Add file function to `async_impl::multipart::{Form, Part}`.

### Notes

Since `From<File> for Body` was added in https://github.com/seanmonstar/reqwest/pull/1360, I decided to use the `stream` function that uses it internally.

[This comment](https://github.com/seanmonstar/reqwest/issues/646#issuecomment-1346431891) says `the backend did not support chunked requests`, but `Transfer-Encoding: chunked` should be accepted for `HTTP/1.1`.

NaokiM03 commented on src/blocking/multipart.rs:
````diff
@@ -104,7 +104,7 @@ impl Form {
     /// ```no_run
     /// # fn run() -> std::io::Result<()> {
-    /// let files = reqwest::blocking::multipart::Form::new()
+    /// let form = reqwest::blocking::multipart::Form::new()
````
Variable names were different from the entities and have been corrected.

NaokiM03 commented on tests/multipart.rs:
```diff
@@ -181,3 +181,61 @@ fn blocking_file_part() {
+        }
+    });
+
+    let url = format!("http://{}/multipart/3", server.addr());
```
I do not know what these numbers mean.
Please let me know if there is something appropriate as I just incremented it.

seanmonstar replied:
They were just arbitrary, they don't mean anything.

Feriixu commented:
@NaokiM03 Are you still developing this? I would very much like this feature merged.

NaokiM03 commented:
@Feriixu 

No, this pull request is open, not draft. This means that the implementation I envisioned has already been done and is not under development.

At the time I wanted this feature, but unfortunately I got no response. So I don't even know if this code is correct.
It would be better to ask one of the committers than to tell me.

seanmonstar reviewed (approved):
Sorry to have let this drop! The code looks excellent to me, starting CI, and assuming all is well, I'd be happy to merge.

seanmonstar commented on tests/multipart.rs:
```diff
@@ -175,3 +175,61 @@ fn blocking_file_part() {
+            assert_eq!(req.headers()["transfer-encoding"], "chunked");
+
+            let mut full: Vec<u8> = Vec::new();
+            while let Some(item) = req.body_mut().next().await {
```
This could be written instead like the test above: `let full = req.collect().await.unwrap().to_bytes();`

NaokiM03 replied:
This caused the test to fail.
Thanks for the tip, the fix was quick. Thank you.

NaokiM03 commented:
The test is down and I will try to fix it.
https://github.com/seanmonstar/reqwest/actions/runs/10588215279/job/29476515082?pr=2106

NaokiM03 commented:
@seanmonstar 
Could you approve again?

Fixed an error that was occurring in CI. https://github.com/seanmonstar/reqwest/pull/2106/commits/350733fb18c29f9517ef0457fe6ebcb885130859
Sorry, I ran all the tests again and found that the doc test was failing, so I fixed it. https://github.com/seanmonstar/reqwest/pull/2106/commits/24b6270b93ce79bf07f0a14198d24d8a495f4347

> Sorry to have let this drop!

No problem at all. I understand that you are busy and I appreciate your contribution to the Rust ecosystem because I like Rust.
